src/main/java/com/synopsys/integration/detect/lifecycle/run/operation/OperationRunner.java

```diff
@@ -390,18 +391,20 @@ public File downloadContainerImage(Gson gson, File downloadDirectory, String con
         return artifactResolver.downloadArtifact(new File(targetPathName), containerImageUri);
     }
 
-    public File getContainerScanImage(Gson gson, File downloadDirectory) throws IntegrationException, IOException, DetectUserFriendlyException {
-        Optional<String> containerImageFilePath = getContainerScanFilePath();
-        File containerImageFile = null;
-        if (containerImageFilePath.isPresent()) {
-            String containerImageUri = containerImageFilePath.get();
-            if (containerImageUri.startsWith("http")) {
-                containerImageFile = downloadContainerImage(gson, downloadDirectory, containerImageUri);
-            } else {
-                containerImageFile = new File(containerImageUri);
+    public File getContainerScanImage(Gson gson, File downloadDirectory) throws OperationException {
+        return auditLog.namedPublic("Retrieve Container Scan Image File", () -> {
+            Optional<String> containerImageFilePath = getContainerScanFilePath();
+            File containerImageFile = null;
+            if (containerImageFilePath.isPresent()) {
```

Comment:
Should there be an error if it is not present?

Reply:
Before this operation is called, there is a validation added [here](https://github.com/blackducksoftware/synopsys-detect/blob/f823395f7a465f341b42bb834964dd1c63522a43/src/main/java/com/synopsys/integration/detect/lifecycle/run/step/ContainerScanStepRunner.java#L55) to check if the property was provided. We skip container scan if the property isn't provided. See #957